service/src/main/java/org/apache/griffin/core/job/JobController.java

```diff
@@ -115,8 +120,19 @@ public JobHealth getHealthInfo() {
     }
 
     @RequestMapping(value = "/jobs/trigger/{id}", method = RequestMethod.POST)
-    @ResponseStatus(HttpStatus.NO_CONTENT)
-    public void triggerJob(@PathVariable("id") Long id) throws SchedulerException {
-        jobService.triggerJobById(id);
+    @ResponseStatus(HttpStatus.OK)
+    public JobInstanceBean triggerJob(@PathVariable("id") Long id, @RequestBody(required = false) String request) throws SchedulerException {
+        return jobService.triggerJobById(id, extractTimeOut(request));
+    }
+
+    private long extractTimeOut(String request) {
```

Comment:
this timeout is rather infra concern, than of particular request, and 99% of all usages won't care about exact value -- let have it as configuration property, not a request body parameter

Reply:
Of course I can remove this param from request and use is only as configuration property. Sometimes there are situations when we need to wait for the creation of JobInstanceBean, but in other situations it is not necessary (see https://github.com/apache/griffin/pull/489#discussion_r266855423)

Reply:
Got it, makes sense. However, let see why exactly it is taking long time.